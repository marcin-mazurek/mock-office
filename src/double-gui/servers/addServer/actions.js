export const INIT = 'addServer/INIT';

export default (name, port, serverType, isSecure, keyPath, certPath) => ({
  type: INIT,
  name,
  port,
  serverType,
  isSecure,
  keyPath,
  certPath
});
