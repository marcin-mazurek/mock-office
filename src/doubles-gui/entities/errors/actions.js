export const ADD = 'errors/ADD';
export const REMOVE = 'errors/REMOVE';

export const add = reason => ({
  type: ADD,
  reason
});

export const remove = id => ({
  type: REMOVE,
  id
});
