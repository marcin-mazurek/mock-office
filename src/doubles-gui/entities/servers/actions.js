export const ADD = 'servers/ADD';
export const SELECT = 'servers/SELECT';
export const START = 'servers/START';
export const STOP = 'servers/STOP';
export const REMOVE = 'servers/REMOVE';
export const RENAME = 'servers/RENAME';

export const add = (name, port, serverType, id, secure) => ({
  type: ADD,
  name,
  port,
  id,
  serverType,
  secure
});

export const select = id => ({
  type: SELECT,
  id
});

export const start = id => ({
  type: START,
  id
});

export const stop = id => ({
  type: STOP,
  id
});

export const remove = id => ({
  type: REMOVE,
  id
});

export const rename = (id, name) => ({
  type: RENAME,
  id,
  name
});
