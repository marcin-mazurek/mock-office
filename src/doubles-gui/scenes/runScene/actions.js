export const RUN = 'scenes/RUN';
export const STOP = 'scenes/STOP';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});

export const stop = (serverId, taskId) => ({
  type: STOP,
  serverId,
  taskId
});
