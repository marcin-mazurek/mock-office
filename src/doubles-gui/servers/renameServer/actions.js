export const INIT = 'renameServer/INIT';

export const init = (id, name) => ({
  type: INIT,
  id,
  name
});
