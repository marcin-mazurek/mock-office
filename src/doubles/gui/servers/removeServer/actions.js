export const INIT = 'removeServer/INIT';

export const init = id => ({
  type: INIT,
  id
});
