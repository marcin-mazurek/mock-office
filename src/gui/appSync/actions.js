export const RESTORE_STATE = 'appSync/RESTORE_STATE';
export const restoreStateAction = servers => ({
  type: RESTORE_STATE,
  servers
});
