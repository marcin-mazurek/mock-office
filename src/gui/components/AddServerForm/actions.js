export const SUBMIT_BUTTON_CLICKED = 'addServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});
