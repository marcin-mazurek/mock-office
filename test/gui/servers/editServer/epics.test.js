/* eslint-disable global-require */
import { fromJS } from 'immutable';
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import { initializedAction } from '../../../../src/gui/servers/editServer/actions';

describe('editServerEpic', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test('actions snapshot on success response', (done) => {
    jest.mock('../../../../src/gui/resources/api', () => ({
      editServer() {
        return Promise.resolve({
          data: {
            id: 'some id',
            port: 3000,
            name: 'name'
          }
        });
      }
    }));
    // eslint-disable-next-line global-require
    const editServerEpic = require('../../../../src/gui/servers/editServer/epics').default;

    const epicMiddleware = createEpicMiddleware(editServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore();
    store.dispatch(initializedAction('some id', fromJS({ port: 3000, name: 'name' })));

    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  it('actions snapshot on failed response', (done) => {
    jest.mock('../../../../src/gui/resources/api', () => ({
      editServer() {
        return Promise.resolve({ error: 'error message' });
      }
    }));
    const editServerEpic = require('../../../../src/gui/servers/editServer/epics').default;

    const epicMiddleware = createEpicMiddleware(editServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore();
    store.dispatch(initializedAction('some id', fromJS({ port: 3000, name: 'name' })));

    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
