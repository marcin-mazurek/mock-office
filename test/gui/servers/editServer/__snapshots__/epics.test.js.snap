// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editServerEpic actions snapshot on failed response 1`] = `
Array [
  Object {
    "id": "some id",
    "type": "editServer/FORM_DID_SUBMIT",
    "values": Object {
      "name": "name",
      "port": 3000,
    },
  },
  Object {
    "reason": "error message",
    "type": "editServer/DID_FAIL",
  },
]
`;

exports[`editServerEpic actions snapshot on success response 1`] = `
Array [
  Object {
    "id": "some id",
    "type": "editServer/FORM_DID_SUBMIT",
    "values": Object {
      "name": "name",
      "port": 3000,
    },
  },
  Object {
    "result": Object {
      "data": Object {
        "id": "some id",
        "name": "name",
        "port": 3000,
      },
    },
    "type": "editServer/DID_SUCCEED",
  },
]
`;
